src/structure.js

```diff
@@ -38,16 +38,17 @@ function Structure (options) {
 inherits(Structure, EventEmitter);
 module.exports = Structure;
 
-Structure.prototype.cursor = function (path) {
+Structure.prototype.cursor = function (path, onChange) {
   var self = this;
   path = path || [];
+  onChange = onChange || function () { };
 
   if (!this.current) {
     throw new Error('No structure loaded.');
   }
 
   var changeListener = function (newRoot, oldRoot, path) {
-
+    onChange(newRoot, oldRoot, path);
```

Comment:
This should go onto its own `changeListener` function. i.e. `changeListener = handleOnChange(this, changeListener);`

Also, `onChange(newRoot, oldRoot, path);` shouldn't be called before `this self.current` is set.